test/test_nn.py

```diff
@@ -515,6 +515,20 @@ def __init__(self):
         self.assertEqual(num_params(n), 3)
         self.assertEqual(num_params(s), 3)
 
+    def test_call_supports_python_dict_output(self):
+        class Net(nn.Module):
+            def __init__(self):
+                super(Net, self).__init__()
+                self.l1 = l
+
+            def forward(self, inputs):
+                return {"output": self.l1(inputs)}
+
+        l = nn.Linear(10, 20)
+        net = Net()
+        data = Variable(torch.randn([5, 10]))
+        net(data)
```

Comment:
This doesn't look like it actually tests the new behavior since there are no backward hooks.